server/jamovi/common/column.cpp

```diff
@@ -616,20 +627,74 @@ double Column::dvalue(int index)
     }
     else // if (dataType() == DataType::TEXT)
     {
-        const char *value = svalue(index);
+        std::string valueStr = svalue(index);
 
-        if (value[0] == '\0')
+        if (valueStr.empty())
         {
             return NAN;
         }
         else
         {
-            double d;
-            char junk;
-            if (sscanf(value, "%lf%1c", &d, &junk) == 1)
-                return d;
-            else
+            if (acceptEuroDecimal && isEuroDecimalPattern(valueStr))
+            {
+                std::replace(valueStr.begin(), valueStr.end(), ',', '.');
+            }
+            try {
+                return std::stod(valueStr);
+            } catch (const std::invalid_argument& e) {
+                return NAN;
+            } catch (const std::out_of_range& e) {
                 return NAN;
+            }
         }
     }
 }
+
+
+/**
+ * Checks if the column's DataType is TEXT and it's values can all be interpretted as 
+ * decimal numbers written in European format.
+ *
+ * @return true if the column contains european-formatted decimals as text values.
+ */
+bool Column::isEuroDecimalTextColumn()
+{
+
+    if (dataType() == DataType::DECIMAL || dataType() == DataType::INTEGER)
+        return false;
```

Comment:
are the above checks redundant, given the immediately following check?

Reply:
Yeah, it mostly there as an artifact of my thought process ...so i'll get rid of it
Also wondering if the `NONE` dt needs to be considered? ...but obviously this is covered by the `if not TEXT` statement as well